src/streamlink/plugins/dlive.py

```diff
@@ -0,0 +1,56 @@
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+from streamlink.compat import unquote_plus
+
+
+QUALITY_WEIGHTS = {
+   "src": 1080,
+}
+
+
+class DLive(Plugin):
+    """
+    Plugin for dlive.tv
+    """
+
+    _url_re = re.compile(r"https?://(?:www\.)?dlive\.tv/")
+    _playback_re = re.compile(r"""(?<=playbackUrl":")(.+?)(?=")""")
+    _livestream_re = re.compile(r""""livestream":null""")
+    _username_re = re.compile(r"(?<=user:)(\w|-)+")
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls._url_re.match(url)
+
+    @classmethod
+    def stream_weight(cls, key):
+        weight = QUALITY_WEIGHTS.get(key)
+        if weight:
+            return weight, "dlive"
+
+        return Plugin.stream_weight(key)
+
+    def _get_streams(self):
+        res = self.session.http.get(self.url)
+
+        playback_url = self._playback_re.search(res.text)
+
+        if playback_url is not None:
+            hls_url = playback_url.group(0)
+            hls_url = bytes(unquote_plus(hls_url), "utf-8").decode(
```

Comment:
does this work correctly with py2 and py3 ?

the facebook addon has something else there

https://github.com/streamlink/streamlink/blob/499301dd2720d9022c051c6004c3efebf217fc12/src/streamlink/plugins/facebook.py#L49-L52

Reply:
I updated and tested it with python2.7 and 3.7. However using ```string_escape``` for the py2 solution did not work. Had to use ```unicode_escape``` for both.